### PR TITLE
Exclude check for ampersand (&) at verbatim fields

### PR DESCRIPTION
**Pull request by koppor**
Fixes https://github.com/koppor/jabref/issues/585#issuecomment-1735461242

Refs https://github.com/JabRef/jabref/issues/8712#issuecomment-1731114692

### Mandatory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/logic/integrity/AmpersandChecker.java [hunk 1]
```diff
@@ -24,6 +26,9 @@ public List<IntegrityMessage> check(BibEntry entry) {
         List<IntegrityMessage> results = new ArrayList<>();
 
         for (Map.Entry<Field, String> field : entry.getFieldMap().entrySet()) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/integrity/AmpersandChecker.java:28]: Maybe even cooler to integrate the filteirng directly in to the getFieldMap call? 
e.g. entry.getFieldMap().entrySet().stream().filter... 
https://mkyong.com/java8/java-8-filter-a-map-examples/

**Comment by koppor** [comment]
@systemoperator Your issue reported at https://github.com/JabRef/jabref/issues/8712#issuecomment-1730481761 should be fixed with this PR.

**Review by Siedlerchr — APPROVED** [review]

File: src/main/java/org/jabref/logic/integrity/AmpersandChecker.java (hunk 1 above)
Comment (koppor) [reply src/main/java/org/jabref/logic/integrity/AmpersandChecker.java]: I tried my best. Needed to use `Tuple2`  for the HtmlChecker. https://github.com/JabRef/jabref/commit/dc82d445c8bbc1ac5126947cb895e062bf6bf6b5

Comment (koppor) [reply src/main/java/org/jabref/logic/integrity/AmpersandChecker.java]: I double checked that all usages do not modify the list any more

File: src/main/java/org/jabref/logic/integrity/AmpersandChecker.java [hunk 2]
```diff
@@ -1,41 +1,43 @@
...
-                // note: when changing the message - also do so in tests
-            }
-        }
-        return results;
+        return entry.getFieldMap().entrySet().stream()
+                    .filter(field -> !field.getKey().getProperties().contains(FieldProperty.VERBATIM))
+                    // We use "flatMap" instead of filtering later, because we assume there won't be that much error messages - and construction of Stream.empty() is faster than construction of a new Tuple2 (including lifting long to Long)
+                    .flatMap(field -> {
+                        // counts the number of even \ occurrences preceding an &
+                        long unescapedAmpersands = BACKSLASH_PRECEDED_AMPERSAND.matcher(field.getValue())
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/integrity/AmpersandChecker.java:30]: move this in it's own method for better readability

File: src/main/java/org/jabref/logic/integrity/AmpersandChecker.java [hunk 3]
```diff
@@ -1,41 +1,43 @@
...
+                                                                               .results()
+                                                                               .map(MatchResult::group)
+                                                                               .filter(m -> CharMatcher.is('\\').countIn(m) % 2 == 0)
+                                                                               .count();
+                        if (unescapedAmpersands == 0) {
+                            return Stream.empty();
+                        }
+                        return Stream.of(new Tuple2<>(field.getKey(), unescapedAmpersands));
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/integrity/AmpersandChecker.java:38]: Java has builtin Pair... That should be sufficient here

**Review by Siedlerchr — CHANGES_REQUESTED** [review]
see comments

**Review by Siedlerchr — APPROVED** [review]

**Review by Siedlerchr — APPROVED** [review]

**Pull request merged**